taiga_contrib_ldap_auth/services.py

```diff
@@ -62,17 +67,29 @@ def ldap_login_func(request):
     # (or any other attribute)
     login_input = request.DATA.get('username', None)
     pass_input = request.DATA.get('password', None)
-
-    # TODO: sanitize before passing to LDAP server?
-    username, email, full_name = connector.login(login = login_input,
-                                                 password = pass_input)
-
-    user = register_or_update(username = username,
-                              email = email,
-                              full_name = full_name)
-
-    data = make_auth_response_data(user)
-    return data
+    try:
+        # TODO: make sure these fields are sanitized before passing to LDAP server!
+        username, email, full_name = connector.login(login = login_input, password = password_input)
+    except connector.LDAPUserLoginError as ldap_error:
```

Comment:
No, you are totally right, I'm a little embarrassed by this mistake TBH. I implemented this due to a user request made over at [ensky/taiga-contrib-ldap-auth](https://github.com/ensky/taiga-contrib-ldap-auth). For my own deployment I've disabled the change password page entirely and I don't use the fallback mechanism. 

I think the correct way to go about this is to ignore the local password entirely *iff* the user exists in the LDAP database. It should, in my opinion, never be possible to override LDAP. The downside is that you would need three separate queries in order to determine whether a user can login:

1. (Optional) Authenticate against LDAP server
2. Check if user exists
3. Check if password matches

I don't think explicitly syncing is a good idea as this kinda defeats the advantages of having LDAP in the first place. Furthermore, you would need to think about what to do if a sync fails (taiga not responding, or whatever). It would become too complex too quickly.

What do you think?